### DISCUSSION TITLE
Is Expo Secure Store on `Android` safe for storing Private Keys?

### DISCUSSION THREAD

**Discussion by map1336** [expo/expo#26695] [Archived: Questions]
Hello Expo Community 👋

I'm building an open-source app for getting familiar with the latest **expo offerings** and i can say that it has been great so far.
https://github.com/ipr0310/my-super-diaries

My next goal for the app is to have a feature of **saving short secrets locally**
So i found `Expo SecureStore` as an option.

But now there is a little doubt i have.
I was reading the docs for `Expo SecureStore`, and i noticed the following:

**On Android**, values are stored in [SharedPreferences](https://developer.android.com/training/data-storage/shared-preferences), encrypted with [Android's Keystore system](https://developer.android.com/training/articles/keystore.html).

<hr/>

The **Android Developers docs**, mention the following:
![image](https://github.com/expo/expo/assets/35440957/521ab1af-7a6d-4549-8ef1-6b7300d87286)
https://developer.android.com/training/data-storage/shared-preferences

As `DataStore` is recommended instead of `SharedPreferences`

<hr/>

**So my final question is:**
For this use case of saving secret passwords, 
Would you still recommend to use `Expo SecureStore` for Android?

**Comment by mcalero11** [comment]
@ipr0310 Do you found an answer?

**Reply by map1336** [reply]
No, but i still used it. Because as it claims the sharedPreferences are encrypted by the device

I guess with the release of newer SDK versions and dropping older android versions, someday it will use the recommended one.
